ext/rinku/autolink.c

```diff
@@ -279,18 +279,23 @@ autolink__url(
 {
 	assert(data[pos] == ':');
 
+	// check that we have "://" and a few characters
 	if ((size - pos) < 4 || data[pos + 1] != '/' || data[pos + 2] != '/')
 		return false;
 
+	// move after the ://
 	link->start = pos + 3;
 	link->end = 0;
 
+  // if there is no domain in this url then stop
 	if (!check_domain(data, size, link, flags & AUTOLINK_SHORT_DOMAINS))
 		return false;
 
+  // find where the url ends
 	link->start = pos;
 	link->end = utf8proc_find_space(data, link->end, size);
 
+  // move to before the protocol
 	while (link->start && rinku_isalpha(data[link->start - 1]))
```

Comment:
... ideally the code would be more readable ... but I was not brave enough to do that :D